### PR TITLE
Reduce gradients before clipping.

### PR DESCRIPTION
**Pull request by taylanbil**
Per https://github.com/pytorch/xla/issues/1708, this is now possible.

### CODE REVIEW

File: fairseq/trainer.py [hunk 1]
```diff
@@ -472,18 +472,17 @@ def maybe_no_sync():
             # normalize grads by sample size
             if sample_size > 0:
                 self.optimizer.multiply_grads(self.args.distributed_world_size / float(sample_size))
+            # tpu-comment: for tpu training, we need to explicitly reduce
+            #   gradients here
+            xm.reduce_gradients(optimizer)
```
Comment (dlibenzi) [inline fairseq/trainer.py]: Before there was some XLA-conditional code.
now you assume here is always XLA.
Is that OK?

Comment (taylanbil) [reply fairseq/trainer.py]: urgh good catch. Thanks. I struggle with this all the time.

**Review by dlibenzi — APPROVED** [review]
Remember to squash!

**Review by dlibenzi — APPROVED** [review]

**Comment by taylanbil** [comment]
I'm now verifying convergence prior to squash+merge'ing this.

**Review by dlibenzi — APPROVED** [review]
You better be ... as this has never ben used this way before 😉

**Comment by taylanbil** [comment]
verified.
